specification/under_development/Base-Profile.md

```diff
@@ -0,0 +1,160 @@
+Attribute groups are numbered such that they can be easily referenced by multiple function definitions or global variables. 
+Arbitrary custom attributes may be optionally attached to any of the declared functions to convey additional information about that function. Consumers of Base Profile compliant programs should hence not rely on the numbering of the entry point attribute group, but instead look for function to which an attribute with the name `EntryPoint` is attached to determine which one to invoke when the program is launched.
+
+To indicate the total number of qubits required to execute the program, a custom attribute with the name `requiredQubits` is defined and attached to the entry point. To indicate the number of registers/bits needed to store measurement results during program execution, a custom attribute with the name `requiredResults` is defined and attached to the entry point. The value of both of these attributes is the string representation of a 32-bit integer constant.
+
+TODO: why the string value, and not an integer?   
+TODO: align naming...
+
+### Function Body
+
+The function body consists of a single block named `entry`.
+In this block, any number of calls to qis functions may be performed.
+To be compatible with the Base Profile these functions must return void. 
+Any arguments to invoke them must be inlined into the call itself; 
+they hence must be constants or a pointer to a [qubit or result]()
+The result of the program execution should be logged using output recording functions.
+
+The following instructions are the *only* LLVM instructions that are permitted within a Base Profile compliant program:
+
+| LLVM Instruction                  | Context and Purpose         | Restrictions for Usage |
+|---------------------------|-------------------|-------------|
+| call    | Used within a function block to invoke any one of the declared qis functions and the output recording functions. | (none) |
+| ret | ... | ... |
+| inttoptr | ... | ... |
+
+## Data Types and Values
+
+Within the base profile, defining local variables is not supported. Arguments to calls correspondingly are expected to be constant values. 
+
+Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*` respectively, where the pointer itself rather than the memory location identifies the qubit or result: For the purpose of passing them as arguments in function calls, a 64-bit signed integer value is cast to the appropriate pointer type. The signed integer that is case must be in the interval `[0, nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and `nrResults` are the required number of qubits and results defined by entry point attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all backends. 
+
+Integers and double-precision floating point numbers are available as in full QIR;
+however, computations using these numeric types are not available.
```

Comment:
But they can be passed as arguments to quantum operations?

Reply:
Yes. I'll flesh out this sentence to make it a short paragraph that should be more explicit about this.